tests/classes/test-class-amp-admanager.php

```diff
@@ -0,0 +1,566 @@
+<?php
+/**
+ * Class Test_AMP_AdManager for test case post type.
+ *
+ * @package AMP_AdManager
+ */
+
+namespace AMP_AdManager\Tests;
+
+use AMP_AdManager\AMP_AdManager;
+
+/**
+ * Unit Test for AMP_AdManager\AMP_AdManager
+ *
+ * @coversDefaultClass \AMP_AdManager\AMP_AdManager
+ *
+ * @group AMP_AdManager
+ *
+ * @package AMP_AdManager
+ */
+class Test_AMP_AdManager extends \WP_UnitTestCase {
+
+	/**
+	 * AMP_AdManager instance
+	 *
+	 * @var AMP_AdManager
+	 */
+	protected $_instance;
+
+	/**
+	 * Setup instance
+	 */
+	public function setup() {
+		parent::setup();
+
+		$this->_instance = new AMP_AdManager();
+	}
+
+	/**
+	 * Test constructor.
+	 *
+	 * @covers ::__construct
+	 */
+	public function test_construct() {
+
+		Utility::invoke_method( $this->_instance, '__construct' );
+
+		$hooks = [
+			[
+				'type'     => 'action',
+				'name'     => 'wp_head',
+				'priority' => 0,
+				'function' => 'load_amp_resources',
+			],
+		];
+
+		// Check if hooks loaded.
+		foreach ( $hooks as $hook ) {
+
+			$this->assertEquals(
+				$hook['priority'],
+				call_user_func(
+					sprintf( 'has_%s', $hook['type'] ),
+					$hook['name'],
+					array(
+						$this->_instance,
+						$hook['function'],
+					)
+				),
+				sprintf( 'AMP_AdManager::__construct() failed to register %1$s "%2$s" to %3$s()', $hook['type'], $hook['name'], $hook['function'] )
+			);
+		}
+
+	}
+
+	/**
+	 * Mock global wp query.
+	 *
+	 * @param array $args WP query arguments.
+	 * @param array $conditions wp query conditions.
+	 */
+	public function mock_wp_query( $args, $conditions ) {
```

Comment:
Can we move this method to Utility class so it can be reused wherever needed

Reply:
sure